stl-decomp-4j/src/main/java/com/github/servicenow/ds/stats/stl/SeasonalTrendLoess.java

```diff
@@ -353,11 +353,17 @@ private void sanityCheck(double[] data) {
 						"SeasonalTrendLoess.Builder: Data series must be at least 2 * periodicity in length");
 
 			if (fPeriodic) {
-				if (fSeasonalWidth != null)
+				int massiveWidth = 100 * data.length;
+
```

Comment:
Note that this can still fail if the subsequent buildSmoother calls are passed data of different length. In practice this tends to be used one-off, which is probably why I never noticed the problem before, but I do sometimes use it in a loop for the same data and this fix helps that case.